proto/src/main/java/com/google/gson/protobuf/ProtoTypeAdapter.java

```diff
@@ -190,6 +191,9 @@ public static Builder newBuilder() {
   private static final com.google.protobuf.Descriptors.FieldDescriptor.Type ENUM_TYPE =
       com.google.protobuf.Descriptors.FieldDescriptor.Type.ENUM;
 
+  private static final ConcurrentMap<String, Map<Class<?>, Method>> mapOfMapOfMethods =
+      new MapMaker().makeMap();
```

Comment:
I have to disagree, since you're shoving them into a static map.  Putting class/method references into a static map is bad smell, because you can end up pinning entire classloaders forever.  Releasing the Gson instance should ultimately allow for cleanup.